test/e2e-samples/specs/custom-login.js

```diff
@@ -12,155 +12,172 @@
+  it('can access user ProfilePage', async () => {
+    await AuthenticatedHomePage.viewProfile();
+    await ProfilePage.waitForPageLoad();
+    expect(await ProfilePage.getEmailClaim()).toBe(params.login.email);
   });
 
-  it('can access resource server messages after login', () => {
+  it('can access resource server messages after login',  async () => {
     // If it's not implicit flow, don't test messages resource server
     if (process.env.TEST_TYPE !== 'implicit') {
       return;
     }
-    authenticatedHomePage.viewMessages();
-    messagesPage.waitForPageLoad();
-    expect(messagesPage.getMessage()).toBeTruthy();
+    await AuthenticatedHomePage.viewMessages();
+    await MessagesPage.waitForPageLoad();
+    expect(await MessagesPage.getMessage()).toBeTruthy();
   });
 
-  it('can log the user out', () => {
-    browser.get(appRoot);
-    authenticatedHomePage.waitForPageLoad();
-    authenticatedHomePage.logout();
-    loginHomePage.waitForPageLoad();
+  it('can log the user out', async () => {
+    await browser.url(appRoot);
+    await AuthenticatedHomePage.waitForPageLoad();
+    await AuthenticatedHomePage.logout();
+    await LoginHomePage.waitForPageLoad();
   });
 
   it('can login with email authenticator', async () => {
     // This test runs only on OIE enabled orgs
-    if (!process.env.ORG_OIE_ENABLED) {
+    if (!exists([
+      process.env.ORG_OIE_ENABLED,
+      process.env.EMAIL_MFA_USERNAME,
+      process.env.PASSWORD,
+      process.env.EMAIL_MFA_USERNAME,
+    ])) {
+      console.warn('Skipping test due to missing env vars');
       return;
     }
 
-    await browser.get(appRoot);
-    await loginHomePage.waitForPageLoad();
+    await browser.url(appRoot);
+    await LoginHomePage.waitForPageLoad();
 
-    await loginHomePage.clickLoginButton();
-    await customSignInPage.waitForPageLoad();
+    await LoginHomePage.clickLoginButton();
+    await OktaSignInPage.waitForPageLoad();
 
-    await customSignInPage.login(process.env.EMAIL_MFA_USERNAME, process.env.PASSWORD);
+    await OktaSignInPage.login(process.env.EMAIL_MFA_USERNAME, process.env.PASSWORD);
 
-    await authenticatorsPage.waitForPageLoad();
-    authenticatorsPage.clickAuthenticatorByLabel('Email');
-    await mfaChallengePage.waitForPageLoad();
+    await AuthenticatorsPage.waitForPageLoad();
+    await AuthenticatorsPage.clickAuthenticatorByLabel('Email');
+    await MFAChallengePage.waitForPageLoad();
 
     // Wait for 5 seconds (default) for email to be received
-    await browser.sleep(process.env.CODE_WAIT_TIME);
+    await browser.pause(process.env.CODE_WAIT_TIME);
 
     // Get the email passcode using ghostinspector email API endpoint
-    await axios.get(`https://email.ghostinspector.com/${process.env.EMAIL_MFA_USERNAME}/latest`).then((response) => {
+    try {
+      const response = await axios.get(`https://email.ghostinspector.com/${process.env.EMAIL_MFA_USERNAME}/latest`);
```

Comment:
In the future, we can replace action like this with util/action function from auth-js e2e, once that submodule is published.